src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -97,40 +160,60 @@ def _read_hdf5_file(
         else:
             ffspec_file = fsspec_file_system.open(file_path, "rb")
 
-        hdf5_file = h5py.File(name=ffspec_file, mode="r")
+        open_file = h5py.File(name=ffspec_file, mode="r")
 
     elif stream_mode == "ros3":
+        import h5py
+
         assert file_path is not None, "file_path must be specified when using stream_mode='ros3'"
 
         drivers = h5py.registered_drivers()
         assertion_msg = "ROS3 support not enbabled, use: install -c conda-forge h5py>=3.2 to enable streaming"
         assert "ros3" in drivers, assertion_msg
-        hdf5_file = h5py.File(name=file_path, mode="r", driver="ros3")
+        open_file = h5py.File(name=file_path, mode="r", driver="ros3")
 
     elif stream_mode == "remfile":
         import remfile
+        import h5py
 
         assert file_path is not None, "file_path must be specified when using stream_mode='remfile'"
         rfile = remfile.File(file_path)
-        hdf5_file = h5py.File(rfile, "r")
+        open_file = h5py.File(rfile, "r")
 
-    elif file_path is not None:
+    elif stream_mode == "zarr":
```

Comment:
I find it a bit confusiong how zarr is a stream mode and a backend at the same time. How does zarr works as  a stream mode?

Reply:
zarr can read directly from cloud buckets! But I see your point. I'll think of a better solution

Reply:
The problem right now is that the streaming for rem and fsspec seem to be coupled to the hdf5 backend which does not need to the case. We should thrown an error when someone passess those streaming methods with zarr and try to fix that later as this PR is very big for this and we probably don't have files for testing this?

Reply:
but how do you know if the remote location is hdf5 or zarr?